android-exoplayer/build.gradle

```diff
@@ -55,7 +55,6 @@ dependencies {
     implementation('com.google.android.exoplayer:extension-okhttp:2.13.3') {
         exclude group: 'com.squareup.okhttp3', module: 'okhttp'
     }
-    implementation 'com.squareup.okhttp3:okhttp:${OKHTTP_VERSION}'
```

Comment:
Would the better fix here be to just change this to double-quotes? We could also do a PR to push this fix upstream, we're not the first to notice it: https://github.com/react-native-video/react-native-video/commit/3dc607c461480a10dce05bad98586444d73487de.

Just for reference: here is the PR adding this line to react-native-video: https://github.com/react-native-video/react-native-video/pull/2340

Reply:
> I actually tried that but it doesn't work as the OKHTTP_VERSION is not available, it only resides within the RN note module. Thus, the build failed when I tried.

Ah good point. We could always wrap this in a `project.hasProperty(...)`, but keeping it simple here by just removing it also seems fine since we obviously haven't needed it.

If you want to go with the "removal" approach you have here, I'm curious if you have any thoughts about possibly commenting out this line and adding a comment describing why it is being commented out. We generally try to keep these forks as close to the source as possible, and I'm thinking it might be hard for someone to backtrack and figure out why we _removed_ this line when the inevitable conflict on this line arises in the future. I don't feel strongly, I'm more just thinking out loud, so let me know what you think.

Reply:
👋 @mchowning @oguzkocer !

Thank you so much for your input! 🙏 

> Ah good point. We could always wrap this in a project.hasProperty(...), but keeping it simple here by just removing it also seems fine since we obviously haven't needed it.

This is a good idea. 👍 

> If you want to go with the "removal" approach you have here, I'm curious if you have any thoughts about possibly commenting out this line and adding a comment describing why it is being commented out. We generally try to keep these forks as close to the source as possible, and I'm thinking it might be hard for someone to backtrack and figure out why we removed this line when the inevitable conflict on this line arises in the future. I don't feel strongly, I'm more just thinking out loud, so let me know what you think.

Yes, this is a good argument, on keeping the forks as close to the source as possible, I agree. 👍 

> It builds without this dependency because okhttp is a transitive dependency of com.facebook.fresco:imagepipeline.

Yes, actually after running `./gradlew dependencies` and checking the before and after result, `com.squareup.okhttp3:okhttp-urlconnection` is also using `okhttp` as a transitive dependency, as the whole `androidx.appcompat:appcompat` dependency itself.

I also did a quick analysis, using `./gradlew dependencies`, a before and after diff for the removal of the explicit `'com.squareup.okhttp3:okhttp:${OKHTTP_VERSION}'` dependency and the only diff I found are the below line, which is found twice on the top level and the `com.facebook.react:react-native:0.66.2` tree:

```
+--- com.squareup.okhttp3:okhttp:${OKHTTP_VERSION} -> 4.9.1 (*)
```

This makes me think that:
1. The `${OKHTTP_VERSION}` is being hardcoded and doesn't change to the `OKHTTP_VERSION=3.12.12` as defined by the `gradle.properties` of the `ReactAndroid` project. Thus, it seems to me that Gradle just ignores while building and then Lint produces the `Incorrect Interpolation` error.
2. With or without this incorrectly defined dependency, due to the `Incorrect Interpolation` error, the end result of building the project is the same. However, this is solely based on building and linting this project individually. If, for some reason, unknown to me due to my lack of React Native experience, when this project is build and used in combination with the `ReactAndroid` project, it might actually do some code-generation magic to substitute this `OKHTTP_VERSION` version with the actual version to produce the final result. Thus, then I would no longer feel comfortable removing this dependency.

> If this project is using okhttp directly, we should keep it and use the version we want. If the project is not using it directly, it should be fine to remove it as whichever dependency needs it should also have it in its pom file.

Good thinking. 🥇 

From what I am seeing this project is using a couple of `okhttp3` imports within the `DataSourceUtil`, which makes me think, just to be on the safe side, since I am not having much React Native experience, to keep this dependency and suppress this `Incorrect Interpolation` Lint error to avoid future such discussions and have Lint success no matter. Wdyt? 🤔

Reply:
👋 @mchowning !

> ...I'd lean toward fixing this with the project.hasProperty(...) wrapper and then switching to using double quotes.

Yes, I think this is better. The only reason I didn't proceed with this change is because you mentioned that we generally try to keep these forks as close to the source as possible and I wasn't sure If I should proceed with this change.

Please expect me applying this change today. 👍

Reply:
This is now done: [b1bff8b](https://github.com/wordpress-mobile/react-native-video/pull/18/commits/b1bff8be70f191719b6c7d4f1e497b1f2e9812d7)